### PR TITLE
Feature/book room

### PR DESCRIPTION
**Pull request by Meekb**
POST function confirmAndPost() line 186 and invoked on line 55 event listener

Feel free to look anything else over and tell me how to write it better or DRYer. 

Thanks Erica!

@Nik

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -17,21 +18,29 @@ function receiveCustProfile(id) {
...
+let sendData = (bookingInfo) => {
+   fetch('http://localhost:3001/api/v1/bookings', {
+        method: 'POST',
+        headers: {'Content-type': 'application/json'},
+        body: JSON.stringify(bookingInfo),
+    })
+
+    .then(response => {
+      console.log(response);
+      if (!response.ok) {
```
Comment (e-spitz) [inline src/apiCalls.js:32]: this is .ok!

File: src/index.js [hunk 2]
```diff
@@ -67,29 +71,32 @@ window.onload = () => {
...
   let username = loginForm.username.value;
   let id = Number(username.split('r')[1]);
   let password = loginForm.password.value;
-  overlook = new Hotel(hotelData);
-  console.log(username, password)
-  if (typeof id === 'number' && id < 51 && password === 'overlook2021') {
+  // validate(id, username, password);
+  console.log(username, password);
+  if (typeof id !== 'number' || id >= 51 || id <= 0 || password !== 'overlook2021') {
+    domUpdates.revealError(loginErr);
```
Comment (e-spitz) [inline src/index.js:81]: love the typeof use here!

**Review by e-spitz — APPROVED** [review]

File: Classes/Hotel.js [hunk 3]
```diff
@@ -17,6 +18,24 @@ class Hotel {
     });
     return search;
   }
+
+  filterOutUnavailable(bkData, rmData, date) {
+    let dateNum = date.split('/');
+    bkData.bookings.forEach(bking => {
+      let compare = bking.date.split('/');
+      if (Number(compare[2]) !== Number(dateNum[1])) {
```
Comment (tarynmartin) [inline Classes/Hotel.js:26]: I could be wrong, but this looks like it's only checking either the date or month, but not both?

File: src/apiCalls.js [hunk 4]
```diff
@@ -17,21 +18,29 @@ function receiveCustProfile(id) {
...
+        throw new Error()
+      } else {
+        return response.json();
+      }
+    })
+    .catch(err => console.log(err, 'here is the pt where error should be displayed'));
```
Comment (tarynmartin) [inline src/apiCalls.js:38]: put in an actual error message to display on the console

**Review by tarynmartin — COMMENTED** [review]
Looks good overall! Just make sure you go back and remove unnecessary console.logs and comments/code that is no longer needed.

**Pull request merged**